torch/fx/experimental/_config.py

```diff
@@ -3,6 +3,8 @@
 from typing import Optional
 
 
+# [@compile_ignored: debug] Fails hard instead of graph breaking on guard on data dependent errors.
+unbacked_strict = os.environ.get("TORCHDYNAMO_UNBACKED_STRICT", "0") == "1"
```

Comment:
1. mm shall the default be 1?
2. i think if we can get away without a global flag and just the default is not graph breaking is even better. 
3. I thought strict mode is the mode where fail at specialization?
TORCHDYNAMO_UNBACKED_STRICT -> TORCHDYNAMO_UNBACKED_GRAPH_BREAK?
i was going to propose mark_unbacked(..., graph_break=False)
but i guess if we end up not able to do (2) above we would need this global flag for non user triggered unbacked?

Reply:
As discussed in chat, global breaks BC.

In general I consider strict a mode where we are more strict, not dissimilar to hack or JavaScript's 'use strict'.

Reply:
I see but since making setting this flag to 1 TORCHDYNAMO_UNBACKED_STRICT=1 does not actually force all unbacked to be strict but only effect the graph break Behaviour. (like it does not make an unbacked variable throw at specialization).
I strongly feel we should not use this name

Reply:
That's a fair point. Ok I renamed it to `no_data_dependent_graph_break`